Scenarios/BlueGreen-Deployment-for-AKS/GAP/bluegreen-aks-solution-content.md

```diff
@@ -0,0 +1,222 @@
+
+![Step3](../media/bg-step3.png)
+
+From a networking perspective this pattern is based on the definition of 3 hostnames:
+- Cluster host - the official hostname used by the consumers of the workloads hosted in the clusters
+- Blue Cluster host - the dedicated host for the blue cluster
+- Green Cluster host - the dedicated host for the green cluster
+
+The cluster host is the one configured at Application Gateway level to manage the ingress traffic, moreover the hostname is also part of the AKS Ingress configuration in order to manage the TLS properly. This host is used only for live transactions and requests.
+
+The Blue and Green cluster hosts are mainly used for:
+- Test and Validate the specific cluster, like mentioned in [Step 2](#t2-sync-k8s-state-between-blue-and-green-cluster)
+- The switch of the live traffic, mainly the hosts are used for the backend pools configuration at Application Gateway level, in this way the switch is transparent for the end user of the workloads.
+For the testing and validation purpose the hosts are also exposed at Application Gateway level with dedicated endpoints and also at AKS Ingress controller level to manage the TLS in the proper way.
+At this stage the validation is based on the infra and app monitoring metrics and official SLO and SLI, when avaialble. If the validation gate is satisfied than is possible to move in the [last state](#t4-blue-cluster-is-destroyed) of the pattern.
+
+
+#### T4: Blue cluster is destroyed
+The execution of the switch of the traffic bring the pattern in the final stage, in which there is still a validation and monitoring that the green cluster is working as expected also with live traffic; is always important to remember that the valadiation and monitoring cover both platform and application level.
+After this validation is completed, then the blue cluster can be destroyed.
+The destruction is a step that is strongly recommended in order to reduce costs and make the proper usage of the elastcity provided by Azure, in particular AKS.
+
+![Step4](../media/bg-step4.png)
+
+
+### Components
+Below the main components and azure services that are part of the blue green deployment for AKS.
+- [Application Gateway](https://azure.microsoft.com/services/application-gateway/), the main responsability is to act as gateway and Load balancer for the AKS clusters.
+- [AKS](https://azure.microsoft.com/services/kubernetes-service/), is the subject of the patter.
+- [Container Registry](https://azure.microsoft.com/services/container-registry/), has the main role to store and distribute the artifacts in the AKS clusters, example of artifacts are Container Images, HELM Charts, etc.. .
+- [Azure Monitor](https://azure.microsoft.com/services/monitor/), is the core monitoring platform for AKS, it is strongly recommended with his native integration with AKS, providing logs, metrics and alerting to manage the different stages of the pattern.
```

Comment:
Consider changing "is the core monitoring platform for AKS, it is strongly recommended with his native integration with AKS, providing logs, metrics and alerting to manage the different stages of the pattern" with "is the core observability platform for AKS and strongly recommended given its native integration with AKS and its ability to provide logging, monitoring and alerting capabilities used to manage the different stages of this pattern."

Reply:
done